### PR TITLE
Added support to delete a plan on a connected account

### PR DESCRIPTION
**Pull request by mrosentr**
Modified plan/client.go Del functions to accept params.
Modified plan.go struct with Customer String.

### CODE REVIEW

**Comment by brandur-stripe** [comment]
@mrosentr Thanks for the pull!

Can you clarify what the new `Customer` parameter is for? I don't think this is going to do anything right now.

**Comment by mrosentr** [comment]
@brandur-stripe Correct, it isn't doing anything right now.  I added it in to be consistent with the api call from the platform based on other functions and also future proofing if needed later on.

**Comment by brandur-stripe** [comment]
> @brandur-stripe Correct, it isn't doing anything right now. I added it in to be consistent with the api call from the platform based on other functions and also future proofing if needed later on.

Okay, I'm not sure I fully follow this. There aren't any plan endpoints that take `customer` right now as far as I'm aware.

You know that usage for Connect would be something more akin to `params.SetAccount()` right?

**Comment by mrosentr** [comment]
Your right.  With no endpoints it doesn't make any sense.  I have removed it.  Thank you.

File: plan/client.go [hunk 1]
```diff
@@ -113,13 +114,24 @@ func (c Client) Update(id string, params *stripe.PlanParams) (*stripe.Plan, erro
...
+		body = &stripe.RequestValues{}
+
+		params.AppendTo(body)
+		commonParams = &params.Params
+	}
+
 	plan := &stripe.Plan{}
-	err := c.B.Call("DELETE", "/plans/"+url.QueryEscape(id), c.Key, nil, nil, plan)
+
+	err := c.B.Call("DELETE", fmt.Sprintf("/plans/%v", id), c.Key, body, commonParams, plan)
```
Comment (brandur-stripe) [inline plan/client.go]: Can we make sure that we keep the `QueryEscape` around here? It looks like it came in for a reason in 98211760beb526dbc58cd91b408488c0ce6929ed.

**Comment by brandur-stripe** [comment]
@mrosentr Thanks! Can you check the test suite run in CI? It looks like there are two more test failures:

```
client/checkin_test.go:126: not enough arguments in call to c.Plans.Del
client/checkin_test.go:171: not enough arguments in call to c.Plans.Del
```

**Comment by mrosentr** [comment]
@brahn-stripe I have corrected the last two test failures.

**Comment by brahn-stripe** [comment]
cc: @brandur-stripe

**Comment by brandur-stripe** [comment]
Thanks!

**Comment by brandur-stripe** [comment]
Released as 23.0.0.

**Pull request merged**